app/models/deploy_group.rb

```diff
@@ -0,0 +1,9 @@
+class DeployGroup < ActiveRecord::Base
+  has_soft_deletion default_scope: true
+
+  belongs_to :environment
+  has_and_belongs_to_many :stages
+
+  validates_presence_of :name, :environment_id
```

Comment:
could validate the presence of the `environment` rather than `environment_id`: http://guides.rubyonrails.org/active_record_validations.html#presence